clang/lib/Driver/ToolChains/BareMetal.h

```diff
@@ -103,7 +112,12 @@ class LLVM_LIBRARY_VISIBILITY StaticLibTool : public Tool {
 
 class LLVM_LIBRARY_VISIBILITY Linker final : public Tool {
 public:
-  Linker(const ToolChain &TC) : Tool("baremetal::Linker", "ld.lld", TC) {}
+  Linker(const ToolChain &TC)
+      : Tool("baremetal::Linker",
+             static_cast<const toolchains::BareMetal &>(TC).isUsingLD()
```

Comment:
Is this argument used for anything? The documentation says

```
  /// The human readable name for the tool, for use in diagnostics.              
  const char *ShortName;
```

Can we just put `"linker"` here like many of the other toolchains instead of getting specific?